src/iap/shared/directives/slide.directive.ts

```diff
@@ -0,0 +1,8 @@
+import { Directive, ViewContainerRef } from '@angular/core';
+
+@Directive({
+	selector: '[slideHost]',
+})
+export class SlideDirective {
+	constructor(public viewContainerRef: ViewContainerRef) {}
+}
```

Comment:
You can implement your carousel component without this directive. You just need to use `#slideHost` template variable and use angular DI in the carousel component to get a needed view:
```typescript
  @ViewChild('slideHost', { read: ViewContainerRef })
  private slideHostView: ViewContainerRef;
```